src/main/client/put.c

```diff
@@ -98,6 +94,8 @@ PyObject *AerospikeClient_Put_Invoke(AerospikeClient *self, PyObject *py_key,
         goto CLEANUP;
     }
 
+    record_initialised = true;
```

Comment:
Two notes:
1. Set the `record_initialised` to true, but you removed the record initalization.  Why was `as_record_init` removed?
2.  The correct spelling is `Initialized`, not `initialised`

Reply:
Here, the variable `as_record rec` was being initialized twice; once when `as_record_init()` is called in this file, and again when `as_record_init_from_pyobject()` is called later in this file. I removed the first one since it's redundant and `as_record_init()` allocates heap memory. The second time `as_record_init()` is called, we lose track of the heap memory from the first `as_record_init()` call. 

The source code for `as_record_init()` -> `as_record_defaults()` shows this behavior